pkg/network/ovn_kubernetes.go

```diff
@@ -1434,7 +1434,7 @@ func shouldUpdateOVNKonPrepull(ovn bootstrap.OVNBootstrapResult, releaseVersion
 // isUserIPsecMachineConfigPresent returns true if user owned MachineConfigs for IPsec plugin
 // are already present in master and worker nodes, otherwise returns false.
 func isUserIPsecMachineConfigPresent(infra bootstrap.InfraStatus) bool {
-	if infra.MasterIPsecMachineConfig == nil && infra.WorkerIPsecMachineConfig == nil {
+	if infra.MasterIPsecMachineConfig == nil || infra.WorkerIPsecMachineConfig == nil {
```

Comment:
Why this change?

Reply:
This is just to avoid nil pointer reference at line.no. 1440 when user rolled out ipsec MachineConfig only in master or worker roles and not on both simultaneously.

Reply:
It also changes the logic. It will return false if either master or worked MCs are not found while line 1440 will return true if any MC does not have the owner ref. Why not
```
return (infra.MasterIPsecMachineConfig != nil && !containsNetworkOwnerRef(infra.MasterIPsecMachineConfig.OwnerReferences)) ||
       (infra.WorkerIPsecMachineConfig != nil && !containsNetworkOwnerRef(infra.WorkerIPsecMachineConfig.OwnerReferences))
```

Reply:
yes @jcaamano , done.